frontend/src/js/common-ui/forms/Form.tsx

```diff
@@ -131,10 +131,11 @@ export const Form = ({
   onSubmit,
   showButtons,
   submitLabel,
-  submitRef
+  submitRef,
+  validateOnSubmitMode = false
```

Comment:
considering the other modes available I would rather pass the mode and default to `onChange` here (and align with RHF)

```suggestion
  mode = 'onChange'
```
and then pass on the mode + adjust the button condition accordingly...